app/views/course/groups/edit.html.slim

```diff
@@ -3,6 +3,19 @@ h2= t('.header')
 = simple_form_for [current_course, @group] do |f|
   = f.input :name
 
-  = f.association :users, collection:  current_course.users
+  table.table
+    thead
+      tr
+        th = t('.name')
+        th = t('.role')
+        th
+    tbody
+      = f.simple_fields_for :group_users do |group_user_form|
+        = render 'group_user_fields', f: group_user_form
+
+  .links
+    = link_to_add_association t('.add_user'), f, :group_users,
```

Comment:
Override the helper and add support for adding a less crazy way of specifying the node and insertion method. Placing the naked JavaScript logic here is not clean.